server/generate.py

```diff
@@ -0,0 +1,187 @@
+import collections
+import datetime
+import functools
+import random
+import string
+
+import names
+
+from server.models import db, User, Course, Assignment, Enrollment, \
+    Backup, Message, Group, Version
+
+original_file = open('tests/files/fizzbuzz_before.py').read()
+modified_file = open('tests/files/fizzbuzz_after.py').read()
+
+def weighted_choice(choices):
+    # http://stackoverflow.com/a/3679747
+   total = sum(w for c, w in choices)
+   r = random.uniform(0, total)
+   upto = 0
+   for c, w in choices:
+      if upto + w >= r:
+         return c
+      upto += w
+   assert False, "Shouldn't get here"
+
+def gen_bool(p=0.5):
+    return random.random() < p
+
+def gen_maybe(value, p=0.5):
+    return value if gen_bool(p) else None
+
+def gen_list(gen, n):
+    return [gen() for _ in range(n)]
+
+def gen_unique(gen, n, attr):
+    generated = collections.OrderedDict()
+    while len(generated) < n:
+        v = gen()
+        generated[getattr(v, attr)] = v
+    return generated.values()
+
+def gen_user():
+    real_name = names.get_full_name()
+    first_name, last_name = real_name.lower().split(' ')
+    return User(
+        name=gen_maybe(real_name, 0.5),
+        email='{}{}{}{}@{}'.format(
+            random.choice([first_name, first_name[0]]),
+            random.choice(string.ascii_lowercase) if gen_bool() else '',
+            random.choice([last_name, last_name[0]]),
+            random.randrange(10) if gen_bool() else '',
+            random.choice(['berkeley.edu', 'gmail.com'])),
+        is_admin=gen_bool(0.05))
+
+def gen_course():
+    return Course(
+        offering='{}/{}/{}{}'.format(
+            'cal',
+            random.choice(['cs61a', 'ds88']),
+            random.choice(['sp', 'su', 'fa']),
+            str(random.randrange(100)).zfill(2)),
+        institution='UC Berkeley',
+        display_name='{} {}{}'.format(
+            random.choice(['CS', 'Data Science']),
+            random.randrange(100),
+            random.choice(['', 'A'])),
+        active=gen_bool(0.3))
+
+def gen_assignment(course):
+    if gen_bool(0.5):
+        display_name = random.choice([
+            'Hog', 'Hog Contest', 'Maps', 'Ants', 'Scheme'
+        ])
+    else:
+        display_name = '{} {}'.format(
+            random.choice(['Homework', 'Lab', 'Quiz']),
+            str(random.randrange(15)).zfill(2))
+    name = course.offering + '/' + display_name.lower().replace(' ', '')
+
+    last_night = datetime.datetime.now().replace(
```

Comment:
I think we want `.utcnow()` - `.now()` uses the computer's local timezone.

The benefit of keeping the seed database in UTC is that we can be consistent in our TZ handling code.